src/rpc_request.rs

```diff
@@ -56,7 +56,7 @@ impl RpcRequest {
            "method": method,
         });
         if let Some(param_string) = params {
-            request["params"] = json!(vec![param_string]);
```

Comment:
This single line change caused a large ripple effect through this PR.  The automatic wrapping of a vector around the param_string meant that we couldn't actually send an RPC request with two or more parameters through rpc_request.rs